components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1116,7 +1155,7 @@ where
         let term = entry.get_term();
         let data = entry.get_data();
 
-        if !data.is_empty() {
+        if (self.peer.is_witness && !can_witness_skip(entry)) || !data.is_empty() {
```

Comment:
I believe it should be `(!self.peer.is_witness || !can_witness_skip(entry)) && !data.is_empty()`.

Reply:
Oh, stupid brain at night